packages/docregistry/src/registry.ts

```diff
@@ -1193,6 +1191,16 @@ namespace DocumentRegistry {
   ];
 }
 
+/**
+ * An interface for a document widget.
+ */
+export
+interface IDocumentWidget<T extends Widget = Widget, U extends DocumentRegistry.IModel = DocumentRegistry.IModel> extends Widget {
+  readonly content: T;
```

Comment:
These should have inline comments (especially "ready").

Reply:
In documenting this promise, I say "A promise resolving when the content widget is ready to be revealed." Would "reveal" be a better name for the promise than "ready"? When the reveal promise resolves, then we reveal the document widget.